src/cpp/src/model_runner.hpp

```diff
@@ -174,9 +228,21 @@ class ModelRunner {
                 Sequence::CPtr sequence = running_sequences[seq_id];
                 for (size_t token_id = 0, position_id = group_position_id; token_id < num_scheduled_tokens; ++token_id, ++position_id, ++gathering_current_index) {
                     // compute token for current sequence
-                    input_ids_data[token_id] = position_id < prompt_len ?
-                        sequence_group->get_prompt_ids()[position_id] :
-                        sequence->get_generated_ids()[position_id - prompt_len];
+                    if (sequence_group_type == SequenceGroupType::TOKENS) {
+                        input_ids_data[token_id] = position_id < prompt_len ?
+                            sequence_group->get_prompt_ids()[position_id] :
+                            sequence->get_generated_ids()[position_id - prompt_len];
+                    } else if (sequence_group_type == SequenceGroupType::EMBEDDINGS) {
+                        auto embeds_pos = position_id > prompt_len ? hidden_size * (position_id - prompt_len) : 0;
+                        for (size_t i = 0; i < hidden_size; i++) {
```

Comment:
I think we can call `std::copy_n` to copy `hidden_size` elements at once

Reply:
Done.